### ISSUE TITLE
Spaces between numbers and units as per NIST standard

### ISSUE THREAD

**Issue by 12ian34** [greshake/i3status-rust#1242]
Hello, I've noticed that since the update with the new formatting, many of the default outputs lack a space between the number and the units. I'm wondering on the reasons behind this choice?

It is a [global standard](https://www.nist.gov/pml/special-publication-811/nist-guide-si-chapter-7-rules-and-style-conventions-expressing-values) to leave a space between a value and its units, plus it's easier (for me, and I presume most others) to read.

I appreciate that it's possible to manually accomplish this by removing the units and then adding a custom unit through Formatting, but this doesn't seem to work with values that have their own units that can change, e.g. those in the net block, and since there's a standard, I'd suggest that leaving a space would be a better default.

**Comment by MaxVerevkin** [comment]
> I'm wondering on the reasons behind this choice?

There are several I can think of:
- to save some space
- having a space before `%` and `°` doesn't make sense
- because of the previous point, it is just technically easier not to have spaces

But I agree, it would be nice to have an option to split the value and prefix/unit. Do you have a syntax for this feature to propose? It should be as easy as possible (formatting string's configuration is already hard to follow).

**Comment by GladOSkar** [comment]
Maybe a `" "` modifier in the unit field, like `{var* b}`?

**Comment by MaxVerevkin** [comment]
> Maybe a " " modifier in the unit field, like {var* b}?

Should it be mutually exclusive with `_` modifier that hides the unit?

**Comment by GladOSkar** [comment]
No i think it's reasonable to use them together. Some people might want to have `12.3 K` without the unit

**Comment by MaxVerevkin** [comment]
@12ian34 what do you think?

I think I'll have time tomorrow to implement this.

**Comment by 12ian34** [comment]
Thanks for looking into this @MaxVerevkin / @GladOSkar - I like the approach you've outlined

**Comment by MaxVerevkin** [comment]
By the way, the space will be added before the prefix, so I think it would make sense to specify the space in prefix' config instead of unit's. Also I hope I'll implement prefix hiding alongside this feature (both require prefix' configuration).

```toml
format = {key; _n*b} # hide the prefix and put a space before the unit
```

```toml
format = {key; n*b} # put a space after the value
```

**Comment by GladOSkar** [comment]
Not sure if hiding the prefix is smart since it's never fixed? But i guess for thing like network speeds one can just assume that the value won't go beyond a certain threshold so I'm okay with it.

**Comment by MaxVerevkin** [comment]
I think
```
format_mem = "{mem_used:1;G*_B}/{mem_total:1;G}"
```
is the smallest we can do with the new formatting. It would render as `1.8G/31GB`.
See https://github.com/greshake/i3status-rust/blob/master/doc/blocks.md#formatting

@MaxVerevkin do you think we can hide the `G` when using `_` as well?

_Originally posted by @GladOSkar in https://github.com/greshake/i3status-rust/issues/1228#issuecomment-840078396_

**Comment by GladOSkar** [comment]
Oops, you got me :D

Still, for Values that change their range a lot this could create issues. But it's most likely fine for things like memory and such :+1:
